ui/src/app/cluster-workflow-templates/cluster-workflow-template-details.tsx

```diff
@@ -47,7 +48,11 @@ export function ClusterWorkflowTemplateDetails({history, location, match}: Route
         [history]
     );
 
-    useEffect(() => setEdited(true), [template]);
+    const resetTemplate = (template: ClusterWorkflowTemplate) => {
+        setTemplate(template);
+        setInitialTemplate(template);
+    };
```

Comment:
```suggestion
    function resetTemplate(template: ClusterWorkflowTemplate) {
        setTemplate(template);
        setInitialTemplate(template);
    }
```

use named functions instead of consts assigned to anonymous functions for better tracing per https://github.com/argoproj/argo-workflows/pull/12062

also helper functions and derivations are conventionally placed before all the `useEffect` calls

Reply:
(same ofc applies to the other changed files as well)

Reply:
Makes sense. Done in 2747d1755c11b49749fb6780bcb2a46f2f4b81e4